compiler/src/iree/compiler/Dialect/HAL/IR/HALBase.td

```diff
@@ -689,6 +689,31 @@ def HAL_DeviceTargetAttr :
     // the `hal.device.targets` attribute is found.
     static SmallVector<DeviceTargetAttr, 4> lookup(Operation *op);
 
+    // Returns true if there is any UnitAttr with |name| in any device
+    // configuration for the given |op|.
+    static bool getConfigAttrAny(Operation *op, StringRef name);
+
+    // Returns the AND of boolean attributes of |name| in all device
+    // configurations found for the given |op|.
+    // Returns nullopt if any config does not have the key defined indicating
+    // that it's not statically known/runtime dynamic.
+    static std::optional<BoolAttr>
```

Comment:
Could this just return an optional bool rather than BoolAttr? So this is a ternary, yes, no, don't know?

Reply:
ah yeah - I had originally written a bunch of template junk where keying on the attr type was important but ripped it out - will fix and do the same for IntegerAttr -> APInt.